sympy/vector/operators.py

```diff
@@ -250,24 +251,52 @@ def gradient(scalar_field, coord_sys=None, doit=True):
     10*R.x*R.z*R.i + 5*R.x**2*R.k
 
     """
-    coord_sys = _get_coord_sys_from_expr(scalar_field, coord_sys)
```

Comment:
` _get_coord_sys_from_expr` return `coordinate_system` when expression has one coordinate system or `None` otherwise. I needed more specific information that just `None`